### ISSUE TITLE
Signal and SMS messages arrived but were lost after power cycle

### ISSUE THREAD

**Issue by montag64** [signalapp/Signal-Android#12734]
- [ x] I have searched open and closed issues for duplicates
- [ x] I am submitting a bug report for existing functionality that does not work as intended
- [ x] I have read https://github.com/signalapp/Signal-Android/wiki/Submitting-useful-bug-reports
- [ x] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description
My phone had about 1% battery when I turned it on this morning. About 5 seconds after I turned it on, it powered off due to low battery, but not before I received at least 4 message notifications from Signal with text (2 encrypted from a Signal user and 2 over SMS). When I powered the phone back on, the messages were completely gone and did not re-arrive so I was never able to fully read them. I had to ask my contacts to resend (of the messages I could see in 5 seconds time, there could have been more I missed). My assumption is this happened because there is some sort of "message received" event or handshake sent back to the Signal backend or the carrier a very short time before the message is actually encrypted and stored on the device, and this "received" should happen only after it is fully stored. I have never seen this before with SMS or any other message protocol. To be clear, the old messages from before this event were still on the phone. I love Signal, but this is a dealbreaker for me from a usability standpoint.

### Steps to reproduce
- Power up phone
- Receive one or more new messages from any user
- Immediately power off phone
- Power on phone and only new messages are gone

**Actual result:** New messages received, phone powered off, but then disappeared after power on
**Expected result:** New messages remain on the phone after being received, OR new messages "re-arrive" a second time after power on (what I have seen with SMS in the past).

### Screenshots
N/A

### Device info
**Device:** Essential Phone PH-1 (Mata)
**Android version:** Android 12 (LineageOS w/ MicroG 19.1 build 19-20221220-microG-mata) *NOT ROOTED*
**Signal version:** 6.7.6

### Link to debug log
I did not get this in time.

**Comment by montag64** [comment]
Why is this getting automatically marked wontfix? Seems like kind of a serious issue?

**Comment by greyson-signal** [comment]
Just to provide context, it's unclear _how_ this could have happened. As Cody alluded to, if notifications are shown, that means that they were written to the database. One thought was that maybe we somehow had a dangling db transaction, but even that couldn't be the case, because the thread that reads messages to present notification is separate from the thread that writes messages, meaning that the transaction must have been completed for the notification thread to read them. 

Worth noting that over the last 1-2 months, we've also been rewriting a lot of our message processing to improve performance, so the code path has changed pretty drastically since your initial report. 

If this has happened again since your initial report, please let us know, and if possible, post a debuglog. Can't do much more investigation without one.
